File: CS/ComboBoxColumnBoundToEnum/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ComboBoxColumnBoundToEnum.Models
{
    public class Order
    {
        public int ID { set; get; }
        public OrderHelper.OrderStatus Status { set; get; }
    }

    public class OrderHelper
    {
        public enum OrderStatus
        {
            Unprocessed = 1,
            Started,
            Sent,
            Finished
        }

        public static List<Order> GetOrders()
        {
            List<Order> orders = new List<Order>();

            for (int i = 1; i <= Enum.GetNames(typeof(OrderStatus)).Length; i++)
            {
                orders.Add(new Order { ID = i, Status = (OrderStatus)i });
            }

            return orders;
        }
    }
}

File: VB/ComboColumnBoundToEnum/Models/Order.vb
Imports ComboColumnBoundToEnum.OrderHelper

Public Class Order
    Public Property ID As Integer
    Public Property Status As OrderStatus
End Class

Public Class OrderHelper
    Public Enum OrderStatus
        Unprocessed = 1
        Started
        Sent
        Finished
    End Enum

    Public Shared Function GetOrders() As List(Of Order)
        Dim orders As List(Of Order) = New List(Of Order)()

        For i As Integer = 1 To [Enum].GetNames(GetType(OrderStatus)).Length
            orders.Add(New Order With {
                .ID = i,
                .Status = CType(i, OrderStatus)
            })
        Next

        Return orders
    End Function
End Class
